stl/inc/xutility

```diff
@@ -5705,12 +5705,12 @@ template <class _Diff, class _Urng>
 class _Rng_from_urng { // wrap a URNG as an RNG
 public:
     using _Ty0 = make_unsigned_t<_Diff>;
-    using _Ty1 = typename _Urng::result_type;
+    using _Ty1 = _Invoke_result_t<_Urng&>;

```

Comment:
Isn't `_Invoke_result_t<_Urng&>` a bit heavyweight when we could simply `decltype(_STD declval<_Urng&>()())`?